.github/workflows/coverage-check.yml

```diff
@@ -0,0 +1,116 @@
+name: Check Test Coverage
+
+on:
+  pull_request:
+    branches: [development]
+
+jobs:
+  test-coverage:
+    name: Check Test Coverage
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout development branch
+        uses: actions/checkout@v3
+        with:
+          ref: development
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18"
+
+      - name: Install dependencies
+        run: sudo apt -y update && sudo apt -y install libusb-1.0-0-dev libudev-dev
+
+      - name: Yarn Install
+        run: yarn install --mode=skip-build && yarn allow-scripts
+
+      - name: Setup Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Setup Forge
+        run: yarn workspace @ubiquity/contracts forge:install
+
+      - name: Get development branch coverage
+        run: |
+          # this is temporary, until we merge the pr branch into development then we can use the yarn workspaces command
+
+          mkdir packages/contracts/scripts/coverage-check && touch packages/contracts/scripts/coverage-check/coverage.ts
+
+          echo 'import { getRPC } from "../runner/rpcutil";
+          import { exec } from "child_process";
+          const setRPCURL = async () => {
+            const optimalRPC = await getRPC();
+            console.log(optimalRPC);
+            return optimalRPC;
+          };
+
+          setRPCURL();' > packages/contracts/scripts/coverage-check/coverage-check.ts    
+
+          RPC_URL=$(npx ts-node packages/contracts/scripts/coverage-check/coverage-check.ts)
+          echo $RPC_URL
+          echo "forge coverage --no-match-path=**/utils/**=**/utils/** --rpc-url $RPC_URL"
```

Comment:
@pwilson77 

have you tried running `forge coverage` without the `--rpc-url` option? 

as far as I understand we don't have any remote tests anymore, only pure unit tests are left, so `forge coverage` should work fine without the `--rpc-url` option

Reply:
> Hi @rndquu could you respond to the earlier message

hi, I can't understand what message I should respond to, could you share a link?

Reply:
was talking of the earlier one  in this thread, where i mentioned that test coverage increased when I added `--rpc-url`

Reply:
> was talking of the earlier one in this thread, where i mentioned that test coverage increased when I added `--rpc-url`

Keep the `--rpc-url` for now, I will check the tests today, perhaps we still have some remote tests (which should be executed on a forked environment)

Reply:
@pwilson77 

We are in the process of removing remote tests (which should be executed on a forked environment). This [PR](https://github.com/ubiquity/ubiquity-dollar/pull/600) removes remote tests completely so you can remove the `--rpc-url` option and do not bother selecting an optimal RPC URL. Running in a local anvil instance should be enough.

@zgorizzo69 pls confirm that after this [PR](https://github.com/ubiquity/ubiquity-dollar/pull/600) is merged we don't have remote tests anymore

Reply:
ok will wait